idalib/src/strings.rs

```diff
@@ -0,0 +1,54 @@
+use std::marker::PhantomData;
+
+use crate::ffi::strings::{
+    build_strlist, clear_strlist, get_strlist_qty, idalib_get_strlist_item_addr,
+    idalib_get_strlist_item_length,
+};
+use crate::ffi::BADADDR;
+
+use crate::idb::IDB;
+use crate::Address;
+
+pub struct StringList<'a> {
+    _marker: PhantomData<&'a IDB>,
+}
+
+impl<'a> StringList<'a> {
+    pub(crate) fn new(_: &'a IDB) -> Self {
+        Self {
+            _marker: PhantomData,
+        }
+    }
+
+    // Note: Seems pretty much useless
+    pub fn build(&self) {
+        unsafe { build_strlist() }
+    }
+
+    // Note: Seems pretty much useless
+    pub fn clear(&self) {
+        unsafe { clear_strlist() }
+    }
+
+    pub fn get_item_addr(&self, index: usize) -> Option<Address> {
```

Comment:
I guess this is and `get_item_length` are low-level primitives so we can just remove the `pub` visibility marker and keep them internal.

Reply:
I think our API should provide a way to get the address of each string in the `StringList`. Also `get_item_length()` might have its use. Not sure about the type that's also shown in the Strings windows of IDA's UI, that's why I haven't implemented it.

I think it would make sense to keep `get_item_addr()` public with a clearer name (do you have any recommedations?) and make `get_item_length()` private. I could also implement a private `get_item_type()` for future use (types are integers that should be interpreted and I didn't look into it yet).

As an alternative, I could modify `get_by_index()` to return an `(Address, String)` tuple. Please advise on your preference.

Reply:
Let’s keep `get_item_addr` as `get_address_by_index`